@here/harp-mapview/lib/DecodedTileHelpers.ts

```diff
@@ -457,26 +458,49 @@ export function applyTechniqueToMaterial(
     level?: number,
     skipExtraProps?: string[]
 ) {
-    Object.getOwnPropertyNames(technique).forEach(propertyName => {
+    const commonProps = Object.getOwnPropertyNames(technique).filter(propertyName => {
         if (
             propertyName.startsWith("_") ||
             BASE_TECHNIQUE_NON_MATERIAL_PROPS.indexOf(propertyName) !== -1 ||
             DEFAULT_SKIP_PROPERTIES.indexOf(propertyName) !== -1 ||
             (skipExtraProps !== undefined && skipExtraProps.indexOf(propertyName) !== -1)
         ) {
-            return;
+            return false;
         }
         const prop = propertyName as keyof typeof technique;
         const m = material as any;
         if (typeof m[prop] === "undefined") {
-            return;
+            return false;
         }
+        return true;
+    });
+
+    // Remove transparent color from the firstly processed properties set.
+    let transparentColor = getTechniqueTransparentColor(technique);
+    if (transparentColor !== undefined) {
+        const idx = commonProps.indexOf(transparentColor);
+        transparentColor = idx >= 0 ? commonProps.splice(idx, 1)[0] : undefined;
```

Comment:
>commonProps.splice(idx, 1)[0]

Huh? Why even searching for this, if we can basically ... check it later as.

```
if (transparentColorPropName !== undefined && transparentColorPropName in technique) {
}
```

Reply:
Yes but we need to remove (by index or by name) this property from the for each loop (of course I wouldn't like to check this condition:
`if propertyName !== transparentColorPropName)` in each loop call, moreover things complicate even more if we want to ignore transparency and opacity values as you suggested in next comments (2 more ifs) . So I guess find and splice is the best we can do with the lack of Array.removeIf utility function.